src/test/java/io/confluent/copycat/hdfs/FailureRecoveryTest.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.copycat.hdfs;
+
+import org.apache.kafka.copycat.data.Schema;
+import org.apache.kafka.copycat.data.Struct;
+import org.apache.kafka.copycat.sink.SinkRecord;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class FailureRecoveryTest extends HdfsSinkConnectorTestBase {
+
+  @Before
+  public void setUp() throws IOException {
+    dfsCluster = false;
+    super.setUp();
+  }
+
+  @Override
+  protected Properties createProps() {
+    Properties props = super.createProps();
+    props.put(HdfsSinkConnectorConfig.STORAGE_CLASS_CONFIG, MemoryStorage.class.getName());
+    props.put(HdfsSinkConnectorConfig.RECORD_WRITER_PROVIDER_CLASS_CONFIG,
+              MemoryRecordWriterProvider.class.getName());
+    return props;
+  }
+
+  @Test
+  public void testCommitFailure() throws Exception {
```

Comment:
Can we validate that we're doing the correct backoff in these tests? Looks like there is a mock context object, so it should be easy to validate.